jest.config.js

```diff
@@ -1,29 +1,70 @@
 const config = {
+  cacheDirectory: '<rootDir>/.cache/jest/',
+  testPathIgnorePatterns: [
+    '/node_modules/',
+    'after-*'
```

Comment:
I think this has broken the `test:build:package` and `test:build:dist` tasks:

```
govuk-frontend on  upgrade-test-jest-percy [$!] via ⬢ v16.17.0 took 20s 
➜ npm run test:build:package      

> test:build:package
> jest tasks/gulp/__tests__/after-build-package.test.js

No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/oliver.byford/Code/govuk-frontend
  1165 files checked across 4 projects. Run with `--verbose` for more details.
Pattern: tasks/gulp/__tests__/after-build-package.test.js - 0 matches

govuk-frontend on  upgrade-test-jest-percy [$!] via ⬢ v16.17.0 took 3s 
➜ npm run test:build:dist   

> test:build:dist
> jest tasks/gulp/__tests__/after-build-dist.test.js

No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/oliver.byford/Code/govuk-frontend
  1165 files checked across 4 projects. Run with `--verbose` for more details.
Pattern: tasks/gulp/__tests__/after-build-dist.test.js - 0 matches
```

The desired behaviour is that the tests in `tasks/gulp/__tests__/` are excluded from running as part of `npm test` but we can call them as part of the `build:package` and `build:dist` tasks to check that the package / dist are what we expect.

(I suspect there's a better way to do this! Maybe we can use the Jest projects config?)

Reply:
Don't worry, this is what code reviews are for 😊 

I'll find a better way of grouping those excluded-but-not-excluded tests—probably Jest config yep!

Reply:
@36degrees I've pushed this up again

The GitHub tests are now split out so Percy only monitors what it needs to:
https://github.com/alphagov/govuk-frontend/blob/upgrade-test-jest-percy/.github/workflows/tests.yml#L47